consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/blockcreation/BftBlockCreator.java

```diff
@@ -46,22 +49,27 @@ public BftBlockCreator(
       final Wei minTransactionGasPrice,
       final Double minBlockOccupancyRatio,
       final BlockHeader parentHeader,
-      final Address miningBeneficiary,
       final BftExtraDataCodec bftExtraDataCodec) {
     super(
         localAddress,
+        miningBeneficiaryCalculator(localAddress, forksSchedule),
         targetGasLimitSupplier,
         extraDataCalculator,
         pendingTransactions,
         protocolContext,
         protocolSchedule,
         minTransactionGasPrice,
-        miningBeneficiary,
         minBlockOccupancyRatio,
         parentHeader);
     this.bftExtraDataCodec = bftExtraDataCodec;
   }
 
+  private static MiningBeneficiaryCalculator miningBeneficiaryCalculator(
+      final Address localAddress, final ForksSchedule<? extends BftConfigOptions> forksSchedule) {
+    return blockNum ->
+        forksSchedule.getFork(blockNum).getValue().getMiningBeneficiary().orElse(localAddress);
```

Comment:
Use the mining beneficiary from the current fork, or else default to the current block creator.